Local users

yoshiemuranaka opened the pull request:
- Delete user working
- Add/edit user working
- No validations or checks

yoshiemuranaka committed:
Updates local user component

Delete user working, dummy modals set up for add/edit
user and settings.

- Renaming components and creating separate
  modal components
- Some updates to button variants to get
  closer to right visual design
- Small updates to grid layout to use container
  with max width set

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Add create and update user functionality

Basic add and edit user working without any form
validation or checks.

- Added plus and settings icons to buttons

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented on src/views/AccessControl/LocalUserManagement/LocalUserManagement.vue:
```diff
@@ -1,83 +1,91 @@
+              <icon-edit />
             </b-button>
             <b-button
+              variant="link"
```
Can we add aria-labels or visually hidden text for the edit and delete buttons in the table?

yoshiemuranaka replied:
Done

derick-montague commented on src/views/AccessControl/LocalUserManagement/ModalUser.vue:
```diff
@@ -0,0 +1,75 @@
+    newUser() {
+      return this.user ? false : true;
+    },
+    form() {
```
Is there a more intuitive name for this function to better describe what it does?

yoshiemuranaka replied:
What do you have in mind..formData()?

derick-montague replied:
I think this is fine. :) Great work!

derick-montague reviewed (commented):
I noticed some errors when trying to edit user name.  Do we want to use the pattern of closing and showing the toast and updating the table on a successful response? I'll add a story to adding toast messages.

gtmills commented on src/views/AccessControl/LocalUserManagement/TableRoles.vue:
```diff
@@ -35,80 +35,80 @@ export default {
           operator: false,
-          user: false,
+          readonly: false,
           callback: false
```
This got renamed to NoAccess. 
Need a master build from the last month to see this. 

curl -k https://${bmc}/redfish/v1/AccountService/Roles
{
  "@odata.context": "/redfish/v1/$metadata#RoleCollection.RoleCollection",
  "@odata.id": "/redfish/v1/AccountService/Roles",
  "@odata.type": "#RoleCollection.RoleCollection",
  "Description": "BMC User Roles",
  "Members": [
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/Administrator"
    },
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/Operator"
    },
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/ReadOnly"
    },
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/NoAccess"
    }

yoshiemuranaka replied:
Done

yoshiemuranaka committed:
Add aria labels to table action buttons

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

gtmills commented on src/store/modules/AccessControl/LocalUserMangementStore.js:
```diff
@@ -19,19 +19,42 @@ const LocalUserManagementStore = {
+      if (privilege) data.RoleId = privilege;
+      if (status !== undefined) data.Enabled = status;
+      api
+        .patch(`/redfish/v1/AccountService/Accounts/${originalUsername}`, data)
```
Awesome stuff! 

This function could also be used to change the password on expired password on login? 
Any thoughts on how we would share it in this case?

yoshiemuranaka replied:
Would be the same as what we have in Angular right? Would just need to reference this store module in the login component when password reset is required

gtmills replied:
Sounds good.

gtmills reviewed (commented):
Nice work! Long term will need a separate view for Operator / readonly roles. These roles can not "Configure users and their accounts", they can however change thier own password.

gtmills commented on src/views/AccessControl/LocalUserManagement/TableRoles.vue:
```diff
@@ -35,80 +35,80 @@ export default {
           callback: false
         },
         {
           description: "Update password for current user account",
```
isn't this a repeat of the one above ?

yoshiemuranaka replied:
Done, thanks for catching

yoshiemuranaka committed:
Update role table Callback to NoAccess

- Remove duplicate table row

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented:
Can you add a story for `No validations or checks`?

yoshiemuranaka committed:
Refactor LocalUserManagementStore

Using dispatch method in context object to dispatch get
instead of referencing the global store (this).

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Update create/edit user role list

Changed Callback to NoAccess in create/edit user
modal.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented:
Integrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28637

derick-montague commented:
Migrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28731/1
